core/src/main/java/org/apache/calcite/util/format/postgresql/PostgresqlDateTimeFormatter.java

```diff
@@ -132,6 +213,10 @@ public class PostgresqlDateTimeFormatter {
           },
           "YYY"),
       new NumberFormatPattern(
+          ChronoUnitEnum.YEARS_IN_CENTURY,
+          0,
+          Integer.MAX_VALUE,
```

Comment:
100?

Reply:
Turns out that PostgreSQL does not limit this. It will accept 4 digit numbers.